vicinity/backends/voyager.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Union
+
+import numpy as np
+from numpy import typing as npt
+from voyager import Index, Space
+
+from vicinity.backends.base import AbstractBackend, BaseArgs
+from vicinity.datatypes import Backend, QueryResult
+from vicinity.utils import Metric, normalize
+
+
+@dataclass
+class VoyagerArgs(BaseArgs):
+    dim: int = 0
+    metric: str = "cosine"
+    ef_construction: int = 200
+    m: int = 16
+
+
+class VoyagerBackend(AbstractBackend[VoyagerArgs]):
+    argument_class = VoyagerArgs
+    supported_metrics = {Metric.COSINE, Metric.EUCLIDEAN}
+    inverse_metric_mapping = {
+        Metric.COSINE: "cosine",
+        Metric.EUCLIDEAN: "l2",
+    }
+
+    metric_int_mapping = {
```

Comment:
I'd index this directly using the `Metric` itself as key. It doesn't make a lot of sense to first convert things to strings, and can only lead to issues (e.g., someone changing "l2" to "euclidean" in one of the mappings, but not the other.)

Reply:
The typing can be fixed by using a `TypeVar` in combination with a Generic argument on the base class. If you don't feel comfortable doing this, let me know, we can add it.

Reply:
I am getting errors(due to `Metric.from_string`) when running `evaluate` method when I use `Metric` as key.

Reply:
But let me see, will figure out, open to discussions and suggestions

Reply:
Leave it like this, it's fine.